aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -599,6 +599,98 @@ Tensor& diff_out(const Tensor& self, int64_t n, int64_t dim, const c10::optional
   }
 }
 
+void axis_check(IntArrayRef& axis, int64_t bound) {
+  std::unordered_set<int64_t> dimensions;
+  for (const auto i : c10::irange(axis.size())) {
+    int64_t pos_axis  = axis[i] >= 0 ? axis[i] : axis.size()+axis[i];
```

Comment:
I don't think this is right because it's using axis.size(), but the correct offset requires the number of dimensions in the tensor. 

Take a look at replacing this function with this one:

https://github.com/pytorch/pytorch/blob/c371542efc31b1abfe6f388042aa3ab0cef935f2/aten/src/ATen/WrapDimUtilsMulti.h#L15

Reply:
Follow-up question, what happens if redundant dimensions are passed to torch.sum()? What function does it use?